src/main/kotlin/org/ptss/support/infrastructure/repositories/CategoryRepository.kt

```diff
@@ -0,0 +1,140 @@
+package org.ptss.support.infrastructure.repositories
+
+import io.quarkus.hibernate.orm.panache.kotlin.PanacheRepository
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.inject.Inject
+import jakarta.persistence.EntityManager
+import jakarta.transaction.Transactional
+import org.ptss.support.domain.interfaces.repositories.ICategoryRepository
+import org.ptss.support.domain.models.Category
+import org.ptss.support.domain.models.Comment
+import org.ptss.support.infrastructure.persistence.entities.CategoryEntity
+import org.ptss.support.infrastructure.persistence.entities.ToolEntity
+import java.time.Instant
+import java.util.UUID
+
+@ApplicationScoped
+class CategoryRepository @Inject constructor(
+    private val entityManager: EntityManager
+) : ICategoryRepository, PanacheRepository<CategoryEntity> {
+    @Transactional
+    override suspend fun getAll(): List<Category> {
+        return entityManager.createQuery(
+            """
+            SELECT DISTINCT c 
+            FROM CategoryEntity c 
+            LEFT JOIN FETCH c.tools
+            """, CategoryEntity::class.java
+        )
+            .resultList
+            .map { it.toDomain() }
+    }
+
+    @Transactional
+    override suspend fun delete(categoryName: String): Category? {
+        val category = findCategoryByName(categoryName) ?: return null
+
+        // Remove tool associations first
+        entityManager.createNativeQuery(
+            "DELETE FROM category_tools WHERE category = :category"
+        )
+            .setParameter("category", categoryName)
+            .executeUpdate()
+
+        // Delete the category
+        entityManager.createNativeQuery(
+            "DELETE FROM categories WHERE category = :category"
+        )
+            .setParameter("category", categoryName)
+            .executeUpdate()
+
+        return category.toDomain()
+    }
+
+    @Transactional
+    override suspend fun create(category: Category): String {
+        val existingTools = if (category.tools.isNotEmpty()) {
+            category.tools.map { tool ->
+                entityManager.find(ToolEntity::class.java, UUID.fromString(tool.id))
+            }
+        } else {
+            emptyList()
+        }
+
+        val categoryEntity = CategoryEntity.fromDomain(category, existingTools)
+
+        // Update bi-directional relationship if there are tools
+        existingTools.forEach { tool ->
+            if (!tool.categories.contains(categoryEntity)) {
+                tool.categories = tool.categories + categoryEntity
+            }
+        }
```

Comment:
wacht huh, dit kan ik verkeerd hebben maar kan een tool meerdere categories hebben? ik dacht dat het een many to one relatie was... Is het dus dat een tool meerdere categorieën kan hebben?

Reply:
Een tool kan idd meerdere categorieën hebben, check swagger doc aangezien het gebruikt wordt als een array